airbyte-webapp/src/components/YamlEditor/YamlEditor.tsx

```diff
@@ -1,25 +1,79 @@
+import { useMonaco } from "@monaco-editor/react";
+import { load, YAMLException } from "js-yaml";
+import { editor } from "monaco-editor/esm/vs/editor/editor.api";
+import { useEffect, useRef, useState } from "react";
+import { useDebounce, useLocalStorage } from "react-use";
+
 import { CodeEditor } from "components/ui/CodeEditor";
 
+import { StreamsListRequestBodyManifest } from "core/request/ConnectorBuilderClient";
 import { useConnectorBuilderState } from "services/connectorBuilder/ConnectorBuilderStateService";
 
 import { DownloadYamlButton } from "./DownloadYamlButton";
 import styles from "./YamlEditor.module.scss";
+import { template } from "./YamlTemplate";
 
 export const YamlEditor: React.FC = () => {
-  const { yamlManifest, setYamlManifest } = useConnectorBuilderState();
+  const yamlEditorRef = useRef<editor.IStandaloneCodeEditor>();
+
+  const [locallyStoredYaml, setLocallyStoredYaml] = useLocalStorage<string>("connectorBuilderYaml", template);
+  const [yamlValue, setYamlValue] = useState(locallyStoredYaml ?? template);
+  useDebounce(() => setLocallyStoredYaml(yamlValue), 500, [yamlValue]);
+
+  const { yamlIsValid, setYamlIsValid, setJsonManifest } = useConnectorBuilderState();
+
+  const monaco = useMonaco();
+
+  useEffect(() => {
+    if (monaco && yamlEditorRef.current && yamlValue) {
+      const errOwner = "yaml";
+      console.log(yamlValue);
+      const yamlEditorModel = yamlEditorRef.current.getModel();
+
+      try {
+        const json = load(yamlValue) as StreamsListRequestBodyManifest;
+        setJsonManifest(json);
+        setYamlIsValid(true);
+
+        // clear editor errors
+        if (yamlEditorModel) {
+          monaco.editor.setModelMarkers(yamlEditorModel, errOwner, []);
+        }
+      } catch (err) {
+        console.log(err.message);
+        if (err instanceof YAMLException) {
+          setYamlIsValid(false);
+          const mark = err.mark;
+          if (yamlEditorModel) {
+            monaco.editor.setModelMarkers(yamlEditorModel, errOwner, [
```

Comment:
One small issue with this is that the error display doesn't really match the current editor theme (it is grey instead of blue). I'm not exactly sure how to customize the theme of those error windows